File: course.py
from sqlalchemy import Column, Integer, String
from postgre import course_map

class Course(course_map):
    __tablename__ = 'course_dim'

    crs_code = Column(String(10), primary_key=True)
    crs_title = Column(String(30))
    fac_code = Column(String(9))
    ins_id = Column(String(9))
    dep_code = Column(String(4))


    def to_json(self):
        dict = {}
        dict["CRS_CODE"] = self.crs_code
        dict["CRS_TITLE"] = self.crs_title
        dict["FAC_CODE"] = self.fac_code
        dict["INS_ID"] = self.ins_id
        dict["DEP_CODE"] = self.dep_code
        return dict

File: src/secondLayer/entity/course.ts
import {
    Column, Entity, JoinColumn, ManyToOne, OneToMany, OneToOne, PrimaryColumn, RelationId
} from 'typeorm';

import ICourse from '../../dataLayer/models/course';
import Faculty from './faculty';

@Entity({ name: "course_dim" })
export default class Course implements ICourse {
  @PrimaryColumn({
    type: "varchar",
    name: "crs_code",
    unique: true,
    length: 10,
  })
  CRS_CODE: string;

  @Column({ type: "varchar", name: "crs_title", unique: true, length: 30 })
  CRS_TITLE: string;

  @Column({ type: "varchar", name: "fac_code", length: 9 })
  FAC_CODE: string;

  @Column({ type: "varchar", name: "ins_id", length: 9 })
  INS_ID: string;

  @Column({ type: "varchar", name: "dep_code", length: 4 })
  DEP_CODE: string;
}
